Merging the initial High Level requirements

acorbin3 opened the pull request:
Adding the initial high level requirements in a separate branch and creating a pull request to be merged into the Math_library branch.

acorbin3 committed:
#1 Adding initial High Level requirements

Adding the initial high level requirements in a separate branch and creating a pull request to be merged into the Math_library branch.

acorbin3 commented on High_Level_Requirements:
```diff
@@ -0,0 +1,5 @@
+Math features
+Sine - This featute shall take an input value and calculate and return the sine value
+Cosine - This featute shall take an input value and calculate and return the sine value
+Floor - TBD
```
This needs to be filled out

acorbin3 reviewed (commented):
Agree to the changes

acorbin3 committed:
#1 Addressing comments in the pull request

acorbin3 commented:
This has been completed but we need someone to review it

acorbin3 commented:
This PR is associated to #1

jlsilvey commented on High_Level_Requirements:
```diff
@@ -0,0 +1,5 @@
+High Level requirements for the math library
+Math features
+Sine - This featute shall take an input value and calculate and return the sine value
```
Feature is mis-spelled. Also we should add some context that allows people to see if there are LLRs associated with it.  Right now, i would say that there needs to be some range bounds defined somewhere (usually in an LLR), but I dont know from looking at this screen if there are LLRs defined.

acorbin3 replied:
- [x] Will update spelling.

From this screen we will only see changes between branches. You could always look a the file by pressing the button "View file" 
In the LLRs I did state that this function shall take in degree's and radians but that was un-bounded.  Should bounds be defined at the HLR level?

jlsilvey replied:
bounds would normally be defined in the LLRs.... its hard to see the full trace from here, and i didnt see the trace in the view file view.... i'm sure you showed me this before but i'm not seeing it now....

acorbin3 replied:
![image](https://user-images.githubusercontent.com/1368121/51768884-8e489d80-20af-11e9-85e1-f2b55ca2f5ca.png)
Here is where to view the file. This is could be helpful if you wanted to see the whole file and if this only showed the Diffs

- [x] I will add comments for the links. 

This isnt exactly how doxygen requirements would be represented. Ideally we would have a links related to this requirement where there was a reference to the LLR's. You would have to manually trace it within the repo or I would envision people would go to the rendered Doxygen to be able to do the tracing.

acorbin3 replied:
Added links but I also had to add unique id's in the LLRs. That change was based on this commit 520fa773f399f4a5ab8eff4c11d170d5c1000a88

acorbin3 replied:
```suggestion
Sine - This feature shall take an input value and calculate and return the sine value
```

jlsilvey commented on High_Level_Requirements:
```diff
@@ -0,0 +1,5 @@
+High Level requirements for the math library
+Math features
+Sine - This featute shall take an input value and calculate and return the sine value
+Cosine - This featute shall take an input value and calculate and return the sine value
```
Feature mis-spelled, also should return cosine value, not sine value.

acorbin3 replied:
- [x] Will update spelling

acorbin3 replied:
- [x] And update requirement to remove the correct value

acorbin3 committed:
#1 Updating spelling and correcting output of 1 requirement

acorbin3 committed:
#1 Update requirements to have links

Updating LLR links based on this commit: 520fa773f399f4a5ab8eff4c11d170d5c1000a88

acorbin3 committed:
#1 {HLR} - Adding unique ID's to HLRs
